TP 2/LEO_2406495616_MadeShandyKrisnanda.py

```diff
@@ -0,0 +1,105 @@
+            print("File not found")
+            genome_file = input("Genome file name: ")
+
+def main_menu(): #Give the user the option to choose what they want to do
+    print("Choose an option:")
+    print("[1] Compute a reverse complement of a k-mer pattern")
+    print("[2] Count a k-mer pattern")
+    print("[3] Find most frequent k-mer patterns")
+    user_option = input("Select an operation [1/2/3]: ") 
+    while user_option not in ["1", "2", "3"]: #Validating input
+        print("Invalid input")
+        user_option = input("Select an operation [1/2/3]: ")
+    return user_option
+
+def reverse_patern(ask_k_mer): #Reverse the pattern according to user input
+    while ask_k_mer: #Validating input
+        for char in ask_k_mer:
+            if char not in ["A", "C", "G", "T"]:
+                print("Invalid input")
+                ask_k_mer = input("Input your pattern: ").upper()
+        break
+    reversed_k_mer=""
+    for char in ask_k_mer[::-1]:
+        if char == "A":
+            reversed_k_mer += "T"
+        elif char == "C":
+            reversed_k_mer += "G"
+        elif char == "G":
+            reversed_k_mer += "C"
+        elif char == "T":
+            reversed_k_mer += "A"
+    return reversed_k_mer
+
+def count_k_mer(genome): #Count the pattern according to user input
+    search_pattern = input("input your pattern: ").upper
+    while search_pattern not in genome: #Validating input
+        print("Pattern not found")
+        search_pattern = input("Input your pattern: ")
+    count_pattern = 0
+    k_mer_reversed = reverse_patern(search_pattern)
+    for i in range(len(genome)-len(search_pattern)+1):
+        if genome[i:i+len(search_pattern)] == search_pattern:
+                count_pattern += 1
+        if genome[i:i+len(search_pattern)] == k_mer_reversed:
+                count_pattern += 1
+    print(count_pattern)
+        
+def frequent_k_mer(genome): #Find the most frequent k-mer
+    len_k_mer = (input("Input your value of k: "))
+    validation_len_k_mer = False
+    while validation_len_k_mer == False: #Validating input
+        try:
+            len_k_mer = int(len_k_mer)
+            validation_len_k_mer = True
+        except:
+            print("Invalid input")
+            len_k_mer = (input("Input your value of k: "))
+    while len_k_mer > len(genome): #Validating input
+        print("k is too large")
+        len_k_mer = int(input("Input your value of k: "))
+    genome_ls = defaultdict(int)
+    for i in range(len(genome)-len_k_mer+1): #Count the k-mer
+            genome_pattern = genome[i:i+len_k_mer]
+            genome_ls[genome_pattern] += 1
+    print(genome_ls)
+    used_k_mer = []
+    for k_mer in genome_ls:
+        if k_mer not in used_k_mer:
+            genome_ls[k_mer] += genome_ls[reverse_patern(k_mer)]
+            genome_ls[reverse_patern(k_mer)] = genome_ls[k_mer]
+            used_k_mer.append(k_mer)
+            used_k_mer.append(reverse_patern(k_mer))   

```

Comment:
instead of adding logic, I think could just call `genome_ls[k_mer] += genome_ls[reverse_patern(k_mer)]` after line 76. In that state. Both of the reverse and non-reverse should have the same value right (there wont be a case where the added non reverse + reverse value appended to the reverse value